### PR TITLE
fix: Improve RegExp for URLs in Admin settings

### LINKED ISSUE #3528: Admin Settings: Improve RegEx for API & FE URL in FE Settings
- Current RegEx implemented in `Admin/Settings` Form doesn't support `localhost` URL, which makes testing of some APIs difficult.
- Improve `RegEx` so that `localhost` or `127.0.0.1/xyz` may be supported.

### PR DESCRIPTION
**Pull request by Reetam-Nandi**
Fixes #3528 

#### Short description of what this resolves:
RegExp for URLs improved to allow acceptance of localhost or 127.0.0.1 URLs in Admin settings.

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by Reetam-Nandi** [comment]
@kushthedude Please review.

File: app/utils/validators.js [hunk 1]
```diff
@@ -20,12 +20,14 @@ export const compulsoryProtocolValidUrlPattern = new RegExp('^(https?:\\/\\/)' /
...
   '^'
+  // localhost inclusion
+  + '(?:(?:localhost)(:\\d{2,5})?|'
   // user:pass authentication
   + '(?:\\S+(?::\\S*)?@)?'
   + '(?:'
   // IP address exclusion
   // private & local networks
-  + '(?!(?:10|127)(?:\\.\\d{1,3}){3})'
+  + '(?!(?:10)(?:\\.\\d{1,3}){3})'
```
Comment (uds5501) [inline app/utils/validators.js]: any link/resource to verify the validity of this check?

Comment (Reetam-Nandi) [reply app/utils/validators.js]: The Loopback IP address ranges from 127.0.0.0 to 127.255.255.255 and as per the Issue the Loopback address(es) should be valid for testing purposes so I removed the 127 address from the exclusion list as it was just excluding any addresses beginning with 10/127.

**Comment by kushthedude** [comment]
@Reetam-Nandi Can you please add a gif or video, showing reg-ex is working fine for all previous rules and new rules ?

**Comment by Reetam-Nandi** [comment]
@kushthedude please check the following [link](https://regex101.com/r/jWpXDV/2) which tests out the RegEx with necessary test cases.

**Pull request merged**